Library/ViewModels/CommentCellViewModel.swift

```diff
@@ -51,30 +54,47 @@ public final class CommentCellViewModel:
       Format.date(secondsInUTC: $0.createdAt, dateStyle: .medium, timeStyle: .short)
     }
 
-    let badge = self.commentUser.signal.skipNil()
-      .map { comment, user in
+    let badge = self.commentUserAndProject.signal.skipNil()
+      .map { comment, user, _ in
         comment.author.id == user?.id.description ? .you : comment.authorBadge
       }
 
     self.authorBadge = Signal.merge(
       badge,
       badge.takeWhen(self.bindStylesProperty.signal)
     )
+
+    let isLoggedOut = self.commentUserAndProject.signal
+      .ignoreValues()
+      .map { _ in AppEnvironment.current.currentUser }
```

Comment:
@moderateepheezy when you originally wrote this you passed the user in to the `configureWith` method but I'm not sure I should be using that or the `currentUser` environmental var?

cc: @justinswart

Reply:
Noticed this too, I guess its' good because it's dependency injected, making it more testable?
But then again we have `withEnvironment` to inject the `User` so maybe it is redundent.

Reply:
@moderateepheezy ^^

Reply:
@msadoon is right about the injection part , but I think we can move it to the CommentCellViewModel so we can inject from there, I only added it to the main vm because that's what I have seen done in the code mostly.

Reply:
Yeah we can rely on our environment for all dependency injection so unless there is a reason I'm not aware of to want to configure and test this cell in isolation from the environment then it is probably not necessary here.

Reply:
Pushed up a commit with the refactor.